### PR TITLE
[MBUILDCACHE-18] Use m-resolver transport layer instead of redefining another one

### PR DESCRIPTION
**Pull request by gnodet**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -142,6 +142,11 @@ under the License.
             <artifactId>maven-resolver-util</artifactId>
             <version>${resolverVersion}</version>
         </dependency>
+        <dependency>
+            <groupId>org.apache.maven.resolver</groupId>
+            <artifactId>maven-resolver-transport-http</artifactId>
+            <version>${resolverVersion}</version>
```
Comment (cstamas) [inline pom.xml:148]: This one is in maven 3.9.x and 4.0.x... so maybe "provided"? Or build cache is meant to work with older versions as well?

File: src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryImpl.java [hunk 2]
```diff
@@ -147,50 +157,33 @@ public void saveArtifactFile( CacheResult cacheResult,
...
-                LOGGER.info( "Cannot download {}, status code: {}", url, statusCode );
-                return Optional.empty();
-            }
-            try ( InputStream content = response.getEntity().getContent() )
-            {
-                return Optional.of( IOUtils.toByteArray( content ) );
-            }
+            GetTask task = new GetTask( new URI( url ) );
+            transporter.get( task );
+            return Optional.of( task.getDataBytes() );
```
Comment (cstamas) [inline src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryImpl.java:165]: Word of warning: unsure the data size here, but if not "heap friendly" sized, this is OOM prone

**Review by cstamas — APPROVED** [review]
LGTM along with OOM remarks (I did not dig into build cache to check these spots)

File: pom.xml (hunk 1 above)
Comment (gnodet) [reply pom.xml:148]: No, build cache requires maven >= 3.9.x, so I'll change it to provided.

**Comment by jira-importer** [comment]
Resolve #248

**Comment by jira-importer** [comment]
Resolve #248

**Pull request merged**